### DISCUSSION TITLE
Apple silicon support

### DISCUSSION THREAD

**Discussion by wpbonelli** [MODFLOW-ORG/modflow6#1286] [General]
Architecture errors have been reported on M1/M2 macs from the distributed `libmf6.dylib`. Can the existing mac build avoid these somehow? An ARM distribution and CI tests could be added once [GitHub-hosted ARM runners](https://github.com/actions/runner-images/issues/2187) come online

**Comment by mbakker7** [comment]
Are there any plans for executables for macOS that run directly on M1/M2 Macs?
Right now it needs Rosetta. It works fine with Rosetta, but not without.

**Comment by wpbonelli** [comment]
Looks like the M1 runner beta [just began](https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/) this week. We could add a build for it, although it's [not clear](https://github.com/actions/runner-images/issues/8439#issuecomment-1743526558) if a free-tier runner will be introduced when the beta ends. I wonder if it's possible/preferable to build a [universal binary](https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary) for mac

**Reply by mbakker7** [reply]
I installed Rosetta which lets you run an intel executable for Mac on an M1/M2 chip. One-line install that works in the background. For people running into this issue, open a terminal and type
```
/usr/sbin/softwareupdate --install-rosetta --agree-to-license     
```
Not sure what an M1/M2 executable would do better. Even faster maybe?

**Reply by wpbonelli** [reply]
Some native ARM vs Intel + rosetta benchmarks could certainly be done.

After some reading / discussion with @jdhughes-usgs it doesn't seem like a universal binary is an option, as those can only be built with Xcode, which doesn't support fortran.

**Reply by wpbonelli** [reply]
Update here: https://github.com/actions/runner-images/issues/8439#issuecomment-1755601587

Sounds like standard-size M1 runners will eventually be free for public projects, but no clear timeline yet

**Reply by wpbonelli** [reply]
M1 runners are now free for public repos

https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/

**Comment by wpbonelli** [comment]
The nightly build [now has a distribution for ARM macs](https://github.com/MODFLOW-USGS/modflow6-nightly-build/releases/tag/20240223). After updating flopy to the tip of develop e.g.

```
pip install https://github.com/modflowpy/flopy/archive/develop.zip
```

It [can be installed](https://github.com/modflowpy/flopy/pull/2115) with

```
get-modflow : --repo modflow6-nightly-build --ostag macarm
```

**Comment by wpbonelli** [comment]
As [GitHub will soon retire the intel mac runners](https://github.blog/changelog/2025-09-19-github-actions-macos-13-runner-image-is-closing-down/) we will have to drop the intel mac distribution, leaving only apple silicon going forward.
